neo/utils.py

```diff
@@ -0,0 +1,856 @@
+        If False, original time stamps are retained.
+        Default is False.
+
+    Keyword Arguments:
+    ------------------
+        Additional keyword arguments used for initialization of the sliced
+        Neo Segment object.
+
+    Returns:
+    --------
+    seg: Neo Segment
+        Temporal slice of the original Neo Segment from t_start to t_stop.
+    """
+    subseg = neo.Segment(**kwargs)
+
+    for attr in [
+            'file_datetime', 'rec_datetime', 'index',
+            'name', 'description', 'file_origin']:
+        setattr(subseg, attr, getattr(seg, attr))
+
+    subseg.annotations = copy.deepcopy(seg.annotations)
+
+    t_shift = - t_start
+
+    # cut analogsignals and analogsignalarrays
+    for ana_id in range(len(seg.analogsignals)):
+        if isinstance(seg.analogsignals[ana_id], neo.AnalogSignal):
+            ana_time_slice = seg.analogsignals[ana_id].time_slice(t_start, t_stop)
+        elif isinstance(seg.analogsignals[ana_id], neo.io.proxyobjects.AnalogSignalProxy):
+            ana_time_slice = seg.analogsignals[ana_id].load(time_slice=(t_start, t_stop))
+        if reset_time:
+            ana_time_slice.t_start = ana_time_slice.t_start + t_shift
+        subseg.analogsignals.append(ana_time_slice)
+
+    # cut spiketrains
+    for st_id in range(len(seg.spiketrains)):
+        if isinstance(seg.spiketrains[st_id], neo.SpikeTrain):
+            st_time_slice = seg.spiketrains[st_id].time_slice(t_start, t_stop)
+        elif isinstance(seg.spiketrains[st_id], neo.io.proxyobjects.SpikeTrainProxy):
+            st_time_slice = seg.spiketrains[st_id].load(time_slice=(t_start, t_stop))
+        if reset_time:
+            st_time_slice = shift_spiketrain(st_time_slice, t_shift)
+        subseg.spiketrains.append(st_time_slice)
+
+    # cut events
+    for ev_id in range(len(seg.events)):
+        if isinstance(seg.events[ev_id], neo.Event):
+            ev_time_slice = event_time_slice(seg.events[ev_id], t_start, t_stop)
+        elif isinstance(seg.events[ev_id], neo.io.proxyobjects.EventProxy):
+            ev_time_slice = seg.events[ev_id].load(time_slice=(t_start, t_stop))
+        if reset_time:
+            ev_time_slice = shift_event(ev_time_slice, t_shift)
+        # appending only non-empty events
+        if len(ev_time_slice):
+            subseg.events.append(ev_time_slice)
+
+    # cut epochs
+    for ep_id in range(len(seg.epochs)):
+        if isinstance(seg.epochs[ep_id], neo.Epoch):
+            ep_time_slice = epoch_time_slice(seg.epochs[ep_id], t_start, t_stop)
+        elif isinstance(seg.epochs[ep_id], neo.io.proxyobjects.EpochProxy):
+            ep_time_slice = seg.epochs[ep_id].load(time_slice=(t_start, t_stop))
+        if reset_time:
+            ep_time_slice = shift_epoch(ep_time_slice, t_shift)
+        # appending only non-empty epochs
+        if len(ep_time_slice):
+            subseg.epochs.append(ep_time_slice)
+
+    return subseg
+
+
+def shift_spiketrain(spiketrain, t_shift):
```

Comment:
This should be a method of the `SpikeTrain` class

Reply:
I agree, all the shift/slice functions should be moved to the respective classes. Should I do that in this PR or in a separate one?

Reply:
If you'd prefer to do that in a separate PR, that's fine with me.